### PR TITLE
Add more docs

### PR DESCRIPTION
**Pull request by Lythenas**
- guide for writing native functions
- change doxygen theme
	- it is now a bit more complicated to generate (but not really since we use a script anyway) but it looks way nicer

### CODE REVIEW

File: src/stdlib.cpp [hunk 1]
```diff
@@ -25,9 +25,21 @@ void Environment::add_default_stdlib() {
     this->add("next", next);
     this->add("select", select);
     this->add("print", print);
+    this->add("force", force);
     this->add("discard_origin", discard_origin);
 }
 
+auto force(const CallContext& ctx) -> CallResult {
```
Comment (tn4799) [inline src/stdlib.cpp:32]: Please add tests for this method

**Review by tn4799 — CHANGES_REQUESTED** [review]

File: include/tree_sitter/tree_sitter.hpp [hunk 2]
```diff
@@ -372,222 +480,231 @@ class Tree;
...
 
     /**
-     * Get the string representation of the type of the node.
+     * @brief The substring of the source code this node represents.
      */
     [[nodiscard]] const char* type() const;
 
     /**
-     * Get the numeric representation of the type of the node.
+     * @brief The numeric TypeId if the node.
```
Comment (tn4799) [inline include/tree_sitter/tree_sitter.hpp:548]:
```suggestion
     * @brief The numeric TypeId of the node.
```

File: include/tree_sitter/tree_sitter.hpp [hunk 3]
```diff
@@ -372,222 +480,231 @@ class Tree;
...
     [[nodiscard]] std::optional<Node> next_sibling() const;
 
     /**
-     * Get the node's previous sibling.
+     * @brief The node's previous sibling.
      *
      * This will also return anonymous nodes.
      *
-     * Can return a null node.
+     * Returns `std::nullopt` this node is already the first sibling.
```
Comment (tn4799) [inline include/tree_sitter/tree_sitter.hpp:641]:
```suggestion
     * Returns `std::nullopt` if this node is already the first sibling.
```

File: include/tree_sitter/tree_sitter.hpp [hunk 4]
```diff
@@ -372,222 +480,231 @@ class Tree;
...
     [[nodiscard]] std::optional<Node> next_named_sibling() const;
 
     /**
-     * Get the node's previous *named* sibling.
+     * @brief The node's previous named sibling.
      *
      * This will not return anonymous nodes.
      *
-     * Can return a null node.
+     * Returns `std::nullopt` this node is already the first named sibling.
```
Comment (tn4799) [inline include/tree_sitter/tree_sitter.hpp:659]:
```suggestion
     * Returns `std::nullopt` if this node is already the first named sibling.
```

File: include/tree_sitter/tree_sitter.hpp [hunk 5]
```diff
@@ -372,222 +480,231 @@ class Tree;
...
 
     /**
-     * Get the start position as a byte offset.
+     * @brief Start position as byte offset.
      */
     [[nodiscard]] std::uint32_t start_byte() const;
 
     /**
-     * Get the end position as a byte offset.
+     * @brief Start position as byte offset.
```
Comment (tn4799) [inline include/tree_sitter/tree_sitter.hpp:669]:
```suggestion
     * @brief End position as byte offset.
```

**Review by tn4799 — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -26,31 +26,37 @@ Alternatively you can use the included scripts:
...
 
-There are two included example applications (possibly out of date). They can be found in the `examples` directory.
-
-- MiniLua-cli: Executes some test strings.
-- MiniLua-gui: Opens a window with an editor and a preview. The editor highlights positions that would be changed due to forcing the values. The preview is completely non-interactive.
-
-For a more up-to-date example take a look at the *interactive_script plugin for rqt*.
+```sh
+# You may need sudo here
+pip3 install jinja2 Pygments
```
[the remark this answers is not in the document]
Comment (Lythenas) [reply README.md:38]: Not sure what you mean by `-operator. They are in a code block.

Also this should work the same on Windows too.

File: include/MiniLua/interpreter.hpp [hunk 7]
```diff
@@ -115,47 +157,53 @@ class Interpreter {
...
 
     /**
-     * Returns the current configuration.
+     * @brief Returns the current configuration.
      */
     auto config() -> InterpreterConfig&;
+    /**
+     * @brief Returns the current configuration.
+     */
     [[nodiscard]] auto config() const -> const InterpreterConfig&;
```
[the remark this answers is not in the document]
Comment (Lythenas) [reply include/MiniLua/interpreter.hpp:166]: Needed? Probably not. Best practice? Definitely yes. Without both a const method you can't use this method in a const function. And without the non-const method you can't do easy modifications like `x.config().all(true)`.

File: include/MiniLua/values.hpp [hunk 8]
```diff
@@ -384,27 +601,52 @@ class Table {
...
 
     // TODO maybe return proxy "entry" type to avoid unnecessary Nil values
+    /**
+     * @brief Access a value by key.
+     */
     auto operator[](const Value&) -> Value&;
+    /**
+     * @brief Access a value by key.
+     */
     auto operator[](const Value&) const -> const Value&;
```
[the remark this answers is not in the document]
Comment (Lythenas) [reply include/MiniLua/values.hpp:630]: Same as above.

File: include/tree_sitter/tree_sitter.hpp [hunk 9]
```diff
@@ -217,77 +305,92 @@ std::ostream& operator<<(std::ostream&, const Edit&);
...
 
     /**
-     * Get the kind of a node type id.
+     * @brief The kind of a node TypeId.
      */
     [[nodiscard]] TypeKind node_type_kind(TypeId) const;
 
     /**
-     * Get the ABI version number for this langauge.
+     * @brief The Tree-Sitter ABI version for this langauge.
```
[the remark this answers is not in the document]
Comment (Lythenas) [reply include/tree_sitter/tree_sitter.hpp:373]: Tree-Sitter calls it ABI in their code. Technically both are correct I think.

**Review by CFU9 — APPROVED** [review]

File: include/tree_sitter/tree_sitter.hpp (hunk 9 above)
[the remark this answers is not in the document]
Comment (tn4799) [reply include/tree_sitter/tree_sitter.hpp:373]: ok. but language is still written wrong.

[the remark this answers is not in the document]
Comment (tn4799) [reply include/tree_sitter/tree_sitter.hpp:373]: I've updated the suggestion.

[the remark this answers is not in the document]
Comment (Lythenas) [reply include/tree_sitter/tree_sitter.hpp:373]: Oh I missed the language typo

**Review by tn4799 — APPROVED** [review]

**Pull request merged**